### PR TITLE
Feat: add movie GraphQL queries

### LINKED ISSUE #9: Create GraphQL query for movies
Movie collection query:
- filtering by title and genre
- expose all properties

### LINKED ISSUE #10: Create GraphQL query for movie (single by id)
Just a single movie by id.
- expose all properties

### PR DESCRIPTION
**Pull request by jesus4497**
### Changes

- Added new 2 queries to get a Movie by its ID and all Movies
- Also did a refactor on how the parse methods were behaving

image

image

Closes #9 and closes #10

### CODE REVIEW

**Review by mimurawil — COMMENTED** [review]
Code looks good. This will need a merge or rebase due to the last commit into main.

I noticed the gql for movies doesn't have the query by search or genre, we probably want to add this so gql doesn't have less functionality than REST APIs

File: src/schema/movie/movie.resolvers.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { DEFAULT_CONTENTFUL_LIMIT } from '../../utils/contentful';
+import getAllMovies from '../../models/Movie/getAll';
+import { CONTENTFUL_INCLUDE } from '../../types/contentful';
+import getMovieById from '../../models/Movie/getById';
+
+export const movieResolvers: Resolvers = {
+	Query: {
+		movie: async (_parent, { id }) => {
+			const movie = await getMovieById(id, {
+				include: CONTENTFUL_INCLUDE.genresWithMovies,
```
Comment (mimurawil) [inline src/schema/movie/movie.resolvers.ts]: We probably just want to use `CONTENTFUL_INCLUDE.genres` here, when querying for the movie(s), we don't want to know the other movies in the same genre of this movie we're querying.

File: src/schema/movie/movie.resolvers.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			}
+			return movie;
+		},
+		movies: async (_parent, { pagination }) => {
+			const perPage = pagination?.perPage || DEFAULT_CONTENTFUL_LIMIT;
+			const page = pagination?.page || 1;
+			const response = await getAllMovies({
+				limit: perPage,
+				page: page,
+				include: CONTENTFUL_INCLUDE.genresWithMovies,
```
Comment (mimurawil) [inline src/schema/movie/movie.resolvers.ts]: Here as well regarding CONTENTFUL_INCLUDE.genres

File: src/schema/movie/movie.resolvers.ts (hunk 1 above)
Comment (jesus4497) [reply src/schema/movie/movie.resolvers.ts]: Yeah, I had doubts here; "how deep should I go with it?"  I agree to stop at genres

Comment (jesus4497) [reply src/schema/movie/movie.resolvers.ts]: Done!

image

**Comment by jesus4497** [comment]
Added the where input to filter by title or genre:

image

File: src/models/Movie/parseMovieData.ts [hunk 3]
```diff
@@ -15,22 +15,33 @@ export function parseMovieSummary(entry: MovieContentfulEntry): MovieSummary {
...
+		summary: fields.summary || '',
+		duration: fields.duration || '',
+		directors: fields.directors || [],
+		mainActors: fields.mainActors || [],
+		datePublished: fields.datePublished || '',
+		rating: fields.rating || '',
+		ratingValue: fields.ratingValue || 0,
+		bestRating: fields.bestRating || 0,
+		worstRating: fields.worstRating || 0,
+		writers: fields.writers || [],
```
Comment (mimurawil) [inline src/models/Movie/parseMovieData.ts]: Do we still need those fallbacks? The types issue got solved in the last merge.

If we still need them, this is breaking the endpoint `GET /movies/titles`

**Review by mimurawil — APPROVED** [review]
🚀

**Pull request merged**